### PR TITLE
feature: 3C-2945 get object legal hold

### PR DESCRIPTION
**Pull request by ilkescality**
Adds route and custom error to GET Object legal hold API.

### CODE REVIEW

File: lib/s3middleware/objectLegalHold.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+/*
+    Format of the xml request:
+    <LegalHold>
+        <Status>ON|OFF</Status>
+    </LegalHold>
+*/
+
+/**
+ * @param {string[]} status - legal hold status parsed from xml to be validated
+ * @return {Error|object} - legal hold status or error
```
Comment (miniscruff) [inline lib/s3middleware/objectLegalHold.js:13]: JSdoc supports type defs or multiple returns for objects, https://jsdoc.app/tags-typedef.html
Might be beneficial here and in other places.
( Won't deny PR though )

File: lib/s3middleware/objectLegalHold.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+    validatedLegalHold.status = validatedStatus.status;
+    return validatedLegalHold;
+}
+
+/**
+ * parse object legal hold - parse and validate xml body
+ * @param {string} xml - xml body to parse and validate
+ * @param {object} log - werelogs logger
+ * @param {function} cb - callback to server
```
Comment (miniscruff) [inline lib/s3middleware/objectLegalHold.js:67]: Callback methods can also be documented with JSDoc https://jsdoc.app/tags-callback.html

File: lib/s3middleware/objectLegalHold.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    });
+}
+
+/**
+ * convert to xml - generates legal hold xml
+ * @param {boolean} legalHold - true if legal hold is on else false
+ * @return {string} - returns legal hold xml
+ */
+function convertToXml(legalHold) {
+    const xml = `<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (miniscruff) [inline lib/s3middleware/objectLegalHold.js:100]: Can just be a return here, but not a big deal

File: lib/s3routes/routes/routeGET.js [hunk 4]
```diff
@@ -111,6 +111,14 @@ function routerGET(request, response, api, log, statsClient, dataRetrievalFn) {
                     return routesUtils.responseXMLBody(err, xml, response, log,
                         corsHeaders);
                 });
+        } else if (request.query['legal-hold'] !== undefined) {
+            // GET object legal hold
```
Comment (miniscruff) [inline lib/s3routes/routes/routeGET.js:115]: Comment can probably be removed given the api method name

**Review by miniscruff — APPROVED** [review]

**Comment by ilkescality** [comment]
duplicate (already merged).

**Pull request closed without merging**